packages/calcite-components/src/components/tile-select/tile-select.tsx

```diff
@@ -319,24 +327,29 @@ export class TileSelect implements InteractiveComponent, LoadableComponent {
           focused,
           [CSS.heading]: Boolean(heading),
           [CSS.headingOnly]: heading && !icon && !description,
-          [CSS.icon]: Boolean(icon),
+          [CSS.icon]: renderIcon,
           [CSS.iconOnly]: !heading && icon && !description,
           [CSS.inputAlignmentEnd]: inputAlignment === "end",
           [CSS.inputAlignmentStart]: inputAlignment === "start",
           [CSS.inputEnabled]: inputEnabled,
-          [CSS.largeVisual]: heading && icon && !description,
+          [CSS.largeVisual]: isLargeVisual,
           [CSS.widthAuto]: width === "auto",
           [CSS.widthFull]: width === "full",
         }}
       >
-        <calcite-tile
-          active={checked}
-          description={description}
-          embed
-          heading={heading}
-          icon={icon}
-          iconFlipRtl={iconFlipRtl}
-        />
+        <div class={{ [CSS.tile]: true, [CSS.tileLargeVisual]: isLargeVisual }}>
+          {icon && (
```

Comment:
If tile and tile-select will share rendering, it may be useful to extract rendering functions.

Reply:
The whole point of this PR is to de-couple them so they're not dependent on each other.  This way we can work exclusively on `calcite-tile` without affecting `calcite-tile-select`.

Reply:
Got it. FWIW, it's still possible to share rendering logic despite the components being decoupled.